### PR TITLE
Get transcript text by ID in AssemblyAI

### PR DESCRIPTION
**Pull request by Swimburger**
If a user already has transcribed an audio or video file, the user can now provide the ID to load the transcript without having to re-transcribe the audio/video file.

Types are now coming from the AssemblyAI node SDK. These type names/contents aren't entirely the same as before.
Do we consider this a breaking change? How do we handle breaking changes?

### CODE REVIEW

File: langchain/src/document_loaders/web/assemblyai.ts [hunk 1]
```diff
@@ -168,37 +151,58 @@ export class AudioTranscriptSentencesLoader extends AssemblyAILoader {
...
+   * @param assemblyAIOptions The options to configure the AssemblyAI loader.
+   * Configure the `assemblyAIOptions.apiKey` with your AssemblyAI API key, or configure it as the `ASSEMBLYAI_API_KEY` environment variable.
+   */
+  constructor(
+    params: CreateTranscriptParameters | string,
+    private subtitleFormat: SubtitleFormat = 'srt',
     assemblyAIOptions?: AssemblyAIOptions
   ) {
-    super(assemblyAIOptions);
+    super(params as string, assemblyAIOptions);
```
Comment (jacoblee93) [inline langchain/src/document_loaders/web/assemblyai.ts:195]: This cast looks weird? Why not make `params: string`?

File: langchain/src/document_loaders/web/assemblyai.ts [hunk 2]
```diff
@@ -1,23 +1,16 @@
...
-  AssemblyAIOptions,
-  CreateTranscriptParams,
-  SubtitleFormat,
-  Transcript,
-  TranscriptSegment,
-} from "../../types/assemblyai-types.js";
+import { AssemblyAIOptions } from "../../types/assemblyai-types.js";
 
-export * from "../../types/assemblyai-types.js";
+export type * from "../../types/assemblyai-types.js";
```
Comment (jacoblee93) [inline langchain/src/document_loaders/web/assemblyai.ts:7]: Hasn't this been deleted below? We should export types from the Client instead right?

**Review by jacoblee93 — COMMENTED** [review]
Thanks for this - a few questions!

**Comment by jacoblee93** [comment]
> If a user already has transcribed an audio or video file, the user can now provide the ID to load the transcript without having to re-transcribe the audio/video file.
> 
> Types are now coming from the AssemblyAI node SDK. These type names/contents aren't entirely the same as before. Do we consider this a breaking change? How do we handle breaking changes?

In this case not much we can do unless you want to make some kind of mapping layer. Overall it looks pretty similar?

File: langchain/src/document_loaders/web/assemblyai.ts (hunk 2 above)
Comment (Swimburger) [reply langchain/src/document_loaders/web/assemblyai.ts:7]: There's one type in that file, and re-exporting everything from the SDK. Should I approach this differently?

File: langchain/src/document_loaders/web/assemblyai.ts (hunk 1 above)
Comment (Swimburger) [reply langchain/src/document_loaders/web/assemblyai.ts]: This is for TypeScript to pick one of the constructor overloads. I merged the overloads into a single constructor for simplicity.

File: langchain/src/document_loaders/web/assemblyai.ts (hunk 2 above)
Comment (jacoblee93) [reply langchain/src/document_loaders/web/assemblyai.ts:7]: Oh my mistake, I thought the file was deleted

**Comment by jacoblee93** [comment]
Thank you!

**Review by Swimburger — COMMENTED** [review]
LGTM!

**Pull request merged**